### ISSUE TITLE
Memory limit not working docker 1.7.0

### ISSUE THREAD

**Issue by enxebre** [ansible/ansible-modules-core#1602]
Hi there
Using docker Client version: 1.7.0 Client API version: 1.19
Using docker module for running a container and set memory limit like:
memory_limit: "256MB"
or:
memory_limit: 256MB

when I docker inspect the container the memory is not set.

**Comment by gregdek** [comment]
cc: @cove @joshuaconner @softzilla @smashwilson

**Comment by bcoca** [comment]
what version of ansible, os, python? could you fill in the template? https://github.com/ansible/ansible/blob/devel/ISSUE_TEMPLATE.md

**Comment by enxebre** [comment]
Hi,
running from os x mavericks 10.9 to ubuntu1404
ansible 1.9.0.1
docker Client version: 1.7.0 Client API version: 1.19
ansible.cfg

```
[defaults]
host_key_checking = no
record_host_keys = no
jinja2_extensions = jinja2.ext.do
timeout = 15
gathering = smart

[ssh_connection]
ssh_args = -o ControlMaster=auto -o ControlPersist=30m
scp_if_ssh = True
pipelining = True
```

trying to:

```
- name: run container
  docker:
    name: sample
    image: "{{sample_image }}"
    state: started
    restart_policy: "{{ sample_restart_policy }}"
    ports:
    - "{{ sample_port }}:{{ sample_port }}"
    expose:
    - "{{ sample_port }}"
    net: "{{ sample_net }}"
    command: "{{ sample_command }}"
    volumes:
    - "/var/run/docker.sock:/tmp/docker.sock"
    memory_limit: "{{ sample_container_memory_limit }}"
```

when docker inspect got:

```
 "Memory": 0,
```

**Comment by ecliptik** [comment]
Same here, added

```
 memory_limit: "512MB"
```

and playbook runs, but docker inspect shows:

```
 "Memory": 0,
 "MemorySwap": 0,
```

**Comment by enxebre** [comment]
Think must be due to I am using docker-py version=1.1.0 as per ansible docs http://docs.ansible.com/ansible/docker_module.html docker-py >= 0.3.0 is a requirement

**Comment by ecliptik** [comment]
I put in a pull request that fixes #1766 due to the wrong data structure reference which was causing container deployments with api v1.19 to break.

On a whim I added,

```
 memory_limit: "512MB"
```

to the playbook, and before where it would deploy successfully and a _docker inspect_ would show _"Memory": 0,_ - it now gives the error,

```
Traceback (most recent call last):
  File "/home/micheal/.ansible/tmp/ansible-tmp-1437509787.17-61083843422347/docker", line 3132, in <module>
    main()
  File "/home/micheal/.ansible/tmp/ansible-tmp-1437509787.17-61083843422347/docker", line 1496, in main
    reloaded(manager, containers, count, name)
  File "/home/micheal/.ansible/tmp/ansible-tmp-1437509787.17-61083843422347/docker", line 1377, in reloaded
    started(manager, containers, count, name)
  File "/home/micheal/.ansible/tmp/ansible-tmp-1437509787.17-61083843422347/docker", line 1354, in started
    created = manager.create_containers(delta)
  File "/home/micheal/.ansible/tmp/ansible-tmp-1437509787.17-61083843422347/docker", line 1227, in create_containers
    containers = do_create(count, params)
  File "/home/micheal/.ansible/tmp/ansible-tmp-1437509787.17-61083843422347/docker", line 1217, in do_create
    result = self.client.create_container(**params)
  File "/usr/lib/python2.6/site-packages/docker/client.py", line 237, in create_container
    volume_driver
  File "/usr/lib/python2.6/site-packages/docker/utils/utils.py", line 546, in create_container_config
    'mem_limit has been moved to host_config in API version 1.19'
docker.errors.InvalidVersion: mem_limit has been moved to host_config in API version 1.19
```

It looks like the method of setting a mem_limit is deprecated, and I poked around where it's coming from, [docker-py/docker/client.py](https://github.com/docker/docker-py/blob/master/docker/client.py#L216), but haven't come up with a fix.

**Comment by nicolaspanel** [comment]
Hi, 
I'm having this error with ansible@1.9.2 / docker@1.8.1
Which *stable* version of ansible should I use to fix?
Best regards

**Comment by achille-roussel** [comment]
Hello, is this fix present in ansible 2.0.0.7 ?

I'm seeing a similar problem, setting `memory_limit: 100MB` on the docker task but the host config still shows `"Memory": 0`.

```
$ python --version
Python 2.7.9
$ ansible-playbook --version
ansible-playbook 2.0.0
  config file = /Users/achille/Work/revolution/ansible.cfg
  configured module search path = Default w/o overrides
```

The ansible task:

``` yaml
- name: setup the {{ docker_image }} container
  docker:
    name: '{{ docker_name }}'
    image: '{{ docker_image }}'
    state: '{{ docker_state }}'
    signal: TERM
    pull: always
    net: host
    detach: yes
    log_driver: json-file
    log_opt:
      max-size: '{{ log_opt_max_size }}'
      max-file: '{{ log_opt_max_file }}'
    volumes:
      - /var/{{ docker_name }}/conf:/var/conf
      - /var/{{ docker_name }}/data:/var/data
    restart_policy: always
    memory_limit: '{{ docker_memory_limit }}'
```

Thanks for your help.
